### PR TITLE
More changes to test IDs

### PR DESCRIPTION
**Pull request by elvisisking**
- see issue #304 
- added `toTestId` function that generates a test ID based on the component name along with a unique name describing what the ID pertains to
- the `toTestId` function uses the converter code from the angular codebase

Here is an example: `connectioncard.student-details-link`. The first part, `connectioncard`, is the component name. The second part, `student`, is the name of the connection. The last part, `details-link`, describes the element whose test ID is defined.

@mmelko Let me know how this works and if you need any IDs changed or added.

### CODE REVIEW

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/1712/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by gashcrumb** [comment]
This seems odd, I checked out the links in the drop-down of the connection card and I see:

image

where's the `n.` coming from?  I don't think we can have `.` in the HTML id attribute if I remember correctly.  Hopefully it didn't come from that slugify function I pointed you to :smirk_cat: if so, :blush: sorry!

**Comment by elvisisking** [comment]
@gashcrumb I put the dot in to separate the component name from the rest of the ID (just seemed more readable). I can take the dots out.

**Comment by gashcrumb** [comment]
Ah, okay, yeah use a `--` or `-` instead, I think dots can cause problems.

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/1721/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/1733/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by gashcrumb** [comment]
I hate to say this but I think the component name's getting compiled out, for example:

image

So every ID has a sorta random letter as a prefix since the id is produced with the minified name I guess.  I don't think that part will work unfortunately, as then the IDs aren't deterministic.

Generally they don't need to be globally unique across the entire app, just to a given page.  So on pages like the connection edit page you probably don't need to be too specific with the name.  Just on things like the connection actions, or integration actions that get used in various pages could do with slightly more specific names, but even those include the object name in them.

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/1802/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

File: app/ui-react/packages/ui/src/utils/testIdGenerator.ts [hunk 1]
```diff
@@ -1,30 +1,31 @@
...
-  if (prefix && uniqueId) {
-    return generateId(prefix) + '--' + generateId(uniqueId);
-  }
+export function toTestId(...values: string[]): string {
+  let testId = '';
 
-  if (prefix) {
-    return generateId(prefix);
-  }
+  for (let i = 0; i < values.length; i++) {
```
Comment (gashcrumb) [inline app/ui-react/packages/ui/src/utils/testIdGenerator.ts]: Couldn't you do `values.map(generateId).join('-')` instead of this `for` loop etc?

Comment (elvisisking) [reply app/ui-react/packages/ui/src/utils/testIdGenerator.ts]: I'll make that change. This changes the segment separator from two dashes to one dash but that is no big deal as we talked about @gashcrumb.

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/1806/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/1812/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Pull request merged**